### ISSUE TITLE
Support for Firefox WebExtensions

### ISSUE THREAD

**Issue by zalmoxisus** [SeleniumHQ/selenium#1181]
Mozilla introduced new [WebExtensions API](https://blog.mozilla.org/addons/2015/08/21/the-future-of-developing-firefox-add-ons/), which works already in Firefox Nightly. The old CPOWs extensions will be deprecated.

Trying to load an extension to Firefox Nightly, Seenium tries to find `install.rdf`, which doesn't exist anymore.

``` javascript
    var firefox = require('selenium-webdriver/firefox');
    var binary = new firefox.Binary('/Applications/FirefoxNightly.app/Contents/MacOS/firefox');
    var profile = new firefox.Profile();
    profile.addExtension('./build/firefox');
    var options = new firefox.Options()
      .setProfile(profile)
      .setBinary(binary);
    this.driver = new firefox.Driver(options);
```

Any plans to support the new extension format for firefox?

**Comment by jleyba** [comment]
We haven't been able been able to get the webdriver.xpi extension (i.e. the FirefoxDriver) signed by Mozilla, and I don't believe we will. This particular feature may have to way for Marionette (Mozilla's WebDriver implementation).

**Comment by cmolocznik** [comment]
Below is a workaround I use written in Java. You just need to copy the entire .xpi to the "[PATH_TO_FIREFOX_PROFILE]/extensions" directory. Firefox will unpack and install automatically when launched via FirefoxDriver.

```java
import org.openqa.selenium.firefox.internal.Extension;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.nio.file.StandardCopyOption;

public class FirefoxWebExtension implements Extension {
    private final File toInstall;

    public FirefoxWebExtension(File toInstall) {
        this.toInstall = toInstall;
    }

    @Override
    public void writeTo(File extensionsDir) throws IOException {
        if (!extensionsDir.exists()) {
            extensionsDir.mkdirs();
        }
        Files.copy(
                toInstall.toPath(),
                Paths.get(
                        extensionsDir.getAbsolutePath(),
                        toInstall.getName()
                ),
                StandardCopyOption.REPLACE_EXISTING
        );
    }
}
```

**Comment by alexristich** [comment]
@cmolocznik I'm new to Selenium, so I'm not too familiar with the project structure. Where would your workaround be placed in order to be functional?

**Comment by tobli** [comment]
See #3846

**Comment by jleyba** [comment]
Fixed by 2f1df982c0905e3594588cdcc902cd2d33aa66c7
